packages/compiler-cli/src/perform_compile.ts

```diff
@@ -79,25 +80,34 @@ export function readConfiguration(
 
           // we are only interested into merging 'angularCompilerOptions' as
           // other options like 'compilerOptions' are merged by TS
-          const existingNgCompilerOptions = {...config.angularCompilerOptions, ...parentOptions};
+          let existingNgCompilerOptions = {...config.angularCompilerOptions, ...parentOptions};
+          if (!config.extends) {
+            return existingNgCompilerOptions;
+          }
 
-          if (config.extends && typeof config.extends === 'string') {
+          const extendsPaths: string[] =
+              typeof config.extends === 'string' ? [config.extends] : config.extends;
+
+          // Call readAngularCompilerOptions recursively to merge NG Compiler options
+          // Reverse the array so the override overrides happen from right to left.
```

Comment:
I'm now noticing some duplication here:

```suggestion
          // Reverse the array so the overrides happen from right to left.
```

Also this would be clearer if the prior sentence ends with a dot.